gobblin-api/src/main/java/org/apache/gobblin/password/EncryptedPasswordAuthenticator.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.password;
+
+import java.net.Authenticator;
+import java.net.PasswordAuthentication;
+import java.util.Properties;
+
+import com.google.common.base.Preconditions;
+
+
+/**
+ * {@link Authenticator} that uses a username and password from the provided {@link Properties} to authenticate, and
+ * also decrypts the password using {@link PasswordManager}
+ */
+public class EncryptedPasswordAuthenticator extends Authenticator {
```

Comment:
Can we add unit tests for this class?

Reply:
Is there anything worth testing here? The only actual logic is calling PasswordManager, and based on PasswordManagerTest, it seems like we can't even enable tests that use that due to flakiness.

Reply:
Added a disabled test that checks an encrypted password, and an enabled test that just uses a password with no encryption.